crawl-ref/source/dat/des/variable/mini_monsters.des

```diff
@@ -2317,6 +2317,66 @@ cccc....cccc
 cccccccccccc
 ENDMAP
 
+NAME:   dreamdust_gnoll_farm
+TAGS:   no_monster_gen transparent
+DEPTH:  D:4-7
+COLOUR: pc = brown
+FTILE:  p = floor_mud
+KMONS:  p = fungus / nothing
+MONS:   gnoll ; scythe, hound
+NSUBST: . = 2=1 / 3=1. / 2 / 1=2. / .
+ITEM:   any potion no_pickup
+MARKER: ! = lua:fog_machine { cloud_type='flame', pow_min = 10, pow_max = 10, \
+                              delay = 10, size = 1, excl_rad = 0 }
+MAP
+..............
+.pppppp.ccccc.
+.pWWWWp.c.!dc.
+.pppppp.c...c.
+.pWWWWp.cc+cc.
+.pppppp.......
+..............
+ENDMAP
+
+NAME:   dreamdust_gnoll_shepherd
+TAGS:   no_monster_gen no_item_gen transparent
+DEPTH:  D:7-11, Lair
+KMONS:  1 = gnoll shaman name:gnoll_shepherd \
```

Comment:
Don't think this is notable enough to warrant a custom monster name/entry. It's just gnoll shaman with yaks or dream sheep. Certainly notable to see dream sheep before Lair, but not to the point where we'd define a custom monster name. It's sort of especially awkward because the shaman isn't a band leader, and it's entirely possible for it to get separated from its "flock". 

If you really like the theme of a gnoll shaman that's a shepherd, I recommend you make some kind of more elaborate "hill" vault with grass etc that somehow contains the flock. Ideally some slightly more elaborate structure to go alongside (a hut or cottage?) and some nice thematic loot. Otherwise I'd just place a gnoll shaman with its custom quarterstaff and its flock and that's fine as a fight for the player to encounter.